src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionEndpoint.cs

```diff
@@ -81,7 +81,11 @@ internal class CodeActionEndpoint : IVSCodeActionEndpoint
 
         cancellationToken.ThrowIfCancellationRequested();
 
-        var delegatedCodeActions = await GetDelegatedCodeActionsAsync(documentContext, razorCodeActionContext, cancellationToken).ConfigureAwait(false);
+        // HTML code actions aren't currently supported in VS Code:
+        // https://github.com/dotnet/razor/issues/8062
+        var delegatedCodeActions = _languageServerFeatureOptions.ProjectConfigurationFileName == LanguageServerConstants.DefaultProjectConfigurationFile
```

Comment:
This feels like the wrong solution to me. IMO we want to minimize client-specific behavior in the LS where possible. Can we have the `CodeActionsHandler` on the VSCode-side always return empty arrays instead (or do so depending on some flag so we can toggle it when we try to turn them on for real)?

If we did want to keep this behavior I'd argue that we should have a better way of checking the client. This seems like it requires a fair bit of deep knowledge that might change.  I'd say we should at least hide it behind something like `__languageServerFeatureOptions.IsVSCode` or `_languageServerFeatureOptions.Client == ClientTypes.VSCode` or something like that.

Reply:
> Can we have the CodeActionsHandler on the VSCode-side always return empty arrays instead (or do so depending on some flag so we can toggle it when we try to turn them on for real)?

`CodeActionsHandler` on the VS Code side only deals with C# code actions. Even if we modified it, we'd still have the issue of HTML code actions coming from `CodeActionEndpoint`.

> I'd say we should at least hide it behind something like __languageServerFeatureOptions.IsVSCode or _languageServerFeatureOptions.Client == ClientTypes.VSCode or something like that.

I can see what you mean, unfortunately besides checking the values of `SingleServerSupport` or `ProjectConfigurationFileName`, I'm not aware of a way to verify if the client is VS Code. We could hide it behind something like `_languageServerFeatureOptions.IsVSCode` as you suggest, but the underlying logic would be exactly the same as it is now (i.e. relying on the values of `SingleServerSupport`/`ProjectConfigurationFileName`).

My perspective is that since this is a temporary solution (i.e. we'd ideally want to support VS Code HTML/C# code actions soon, especially since the latter purportedly used to work), this workaround seems appropriate for now since I'd say it's unlikely that we'll change our configuration file names in the near future.

Reply:
I agree with Ryan that this feels wrong, but personally, I don't think `_languageServerFeatureOptions.IsVSCode` is any better than this. I would prefer `_languageServerFeatureOptions.SupportsDelegatedCodeActions`. ie, describe the behaviour that is being disabled, not the environment in which it is being disabled.

That also gives us a very clear list of things that need to be fixed when we do add that support, as we just remove the property, and the compiler will ensure we didn't miss a sneaky `if` somewhere, or a strange mock configuration.

This might be temporary, but I could also see us being randomized by bugs once we get a release out, after all this time, that it might not be as temporary as you'd hope.

Reply:
That's fair, I could see how having `SupportsDelegatedCodeActions` would leave us less bug prone in the future. I've added the property in my latest push.